scripts/basic-workflow-tests.sh

```diff
@@ -24,13 +24,21 @@ BOLD='\033[1m'
 NC='\033[0m' # No Color
 
 SCRIPTDIR=$(dirname "$0")
+E2E_TEST_DIR="$(pwd)/__e2e_test"
+export XDG_CONFIG_HOME="${E2E_TEST_DIR}/.config"
+export XDG_DATA_HOME="${E2E_TEST_DIR}/.local/share"
+
```

Comment:
Let's also define the third and last XDG home directory in case we need it later.
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

```suggestion
export XDG_STATE_HOME="${E2E_TEST_DIR}/.local/share"

```

Reply:
What's the difference between this and `XDG_DATA_HOME`? 
Also - I can replace this from being "XDG_*" to just being DATA, CONFIG, etc., whatever we need here if you'd like.

Reply:
The state home is for less important data than data home. Application directory standards have multiple levels of importance and often different directories for local and roaming data.
Files in data home should be backed up and synchronized between machines. Files in state home are less important and not portable (local, not synced).

Reply:
Gotcha. I've defined all 3 and have the scripts using them.